chainer/functions/theano/theano_function.py

```diff
@@ -0,0 +1,194 @@
+import collections
+
+import numpy
+import six
+
+try:
+    import theano
+    import theano.sandbox.cuda as theano_cuda
+    _available = True
+except ImportError:
+    _available = False
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def _to_var_tuple(vs):
+    msg = ('inputs and outputs must be a TensorVariable, a list '
+           'of TensorVariable or a tuple of TensorVariable')
+
+    if isinstance(vs, theano.tensor.TensorVariable):
+        return vs,
+    elif isinstance(vs, collections.Iterable):
+        vs = tuple(vs)
+        if not all(isinstance(v, theano.tensor.TensorVariable) for v in vs):
+            raise TypeError(msg)
+        return vs
+    else:
+        raise TypeError(msg)
+
+
+class TheanoFunction(function.Function):
+
+    """Theano function wrapper.
+
+    This function wrapps Theano function as a :class:`chainer.Function`.
+    A user needs to make input Theano variables and output Theano variables.
+    This function automatically creates Theano function for forward calculation
+    and backward calculation from inputs and ouptuts. And then, it sends data
+    in :class:`chainer.Variable` to the function and gets results from Theano.
+
+    If a user want to use GPUs, he/she can directly send GPU data to Theano
+    function without copying.
+
+    .. admonition:: Example
+
+       >>> x = theano.tensor.fvector()
+       >>> y = theano.tensor.fvector()
+       >>> z = x + y
+       >>> w = x - y
+       >>> f = F.TheanoFunction(inputs=[x, y], outputs=[z, w], gpu=False)
```

Comment:
It looks like `gpu` has been changed to `optimize_gpu`. Please update documentation.